excavator/utils.py

```diff
@@ -80,7 +81,49 @@ def env_int(name, required=False, default=empty):
     value = get_env_value(name, required=required, default=default)
     if value is empty:
         raise ValueError(
-            "`env_int` requires either a default value to be specified, or, for "
+            "`env_int` requires either a default value to be specified, or for "
             "the variable to be present in the environment"
         )
     return int(value)
+
+
+def _parse_timestamp(timestamp_str):
+    timestamp = float(timestamp_str)
+    return datetime.datetime.fromtimestamp(timestamp)
+
+
+def _parse_iso8601(iso8601_str):
+    try:
+        import iso8601
+    except ImportError:
+        raise ImportError(
+            'Parsing iso8601 datetime strings requires the iso8601 library'
+        )
+    return iso8601.parse_date(iso8601_str)
+
+
+_formats = {
+    'timestamp': _parse_timestamp,
+    'iso8601': _parse_iso8601,
+}
+
+
+def env_datetime(name, default=empty, required=False, fmt='timestamp'):
```

Comment:
> If you are returning a datetime object for env_datetime, do you need dateutils? 

How do you parse an arbitrary string into a python `datetime`.  `python-dateutil` is the most reliable way I know to do that.  Alternatively, I could use [`strptime`](https://docs.python.org/2/library/datetime.html#datetime.datetime.strptime) but that requires a format to be specified which is not entirely user friendly.  

Maybe I could allow `fmt` as an optional keyword.  If provided, then use `strptime`, otherwise fall back to trying to use `python-datetuils` (and throw a friendly error if it isn't installed cause I don't want to add any dependencies.)